src/services/authRequest.ts

```diff
@@ -1,16 +1,16 @@
 import request from '@/utils/request';
 import { AuthCredentials } from '@/constants/types';
 
-const authRequest = async <T>(endpoint: string, credentials: AuthCredentials): Promise<T> => {
-  const { data } = await request<T>(`${process.env.NEXT_PUBLIC_API}/users/${endpoint}`, {
+const authRequest = async (endpoint: string, credentials: AuthCredentials) => {
+  const { data, response } = await request(`${process.env.NEXT_PUBLIC_API}/users/${endpoint}`, {
     method: 'POST',
     body: JSON.stringify(credentials),
     headers: {
       'Content-Type': 'application/json',
     },
```

Comment:
remove headers from requests and use in the request file itself. This way you wouldn't have to repeat it every time.